### ISSUE TITLE
Post command is never executed

### ISSUE THREAD

**Issue by deadlocker8** [primaeval/plugin.video.iptv.recorder#23]
Any simple post command set in the plugin settings page is never executed. I have tested a simple bash script on LibreElec and a simple batch script on windows each only creating a basic file. It seems that the ffmpeg process is never stopped thus the infinite loop in the created job py-file is never exited properly and the post command therefore never executed.

##### Windows 10
- Windows 10 (1809)
- Kodi 18.1 (Git: 20190217-8cfdc895f3)
- IPTV-Recorder Plugin: 0.0.122

##### LibreELEC
- LibreELEC Offical 9.0.1 (Kernel 4.19.23)
- Kodi (18.1 newclock5_18.1-LEIA) Compile Date: 21.02.19
- IPTV-Recorder Plugin: 0.0.122

**Comment by primaeval** [comment]
Thanks. Try version 0.0.125.
Make sure the path is accessible to your post command on the os command line. It doesn't go through Kodi like the ffmpeg recording cmd does.

**Comment by deadlocker8** [comment]
Thanks for your fast reply.
It's now working on the LibreELEC system using the 0.0.125 plugin version. However on Windows the ffmpeg process (or rather the parent cmd.exe process) is never killed, so ffmpeg will continue to write data to stdout and the loop will therefore run infinitly. If I manually stop the parent cmd.exe process or the ffmpeg.exe process the loop is exiting correctly and the post command is executed.

**Comment by primaeval** [comment]
Have you re-created your rules and jobs?
If you have you should have the extra code that checks for no data and exits the while loop.
This works and runs notepad after it finishes recording.

```
# -*- coding: utf-8 -*-
import os, subprocess, time
import xbmc,xbmcvfs,xbmcgui
xbmcgui.Dialog().notification("Recording: BBC News HD", "BBC News", sound=True)
cmd = ['D:\\utils\\ffmpeg.exe', '-i', u'http://******', '-y', '-t', '199', '-c', 'copy', '-f', 'mpegts', '-']
p = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
f = open(r'D:\kodi-18.1-Leia-x64\portable_data\userdata\addon_data\plugin.video.iptv.recorder\jobs\23da3f8f-53c1-11e9-8faf-5404a6420d2f.py.pid', 'w+')
f.write(repr(p.pid))
f.close()
video = xbmcvfs.File(r"D:\kodi-18.1-Leia-x64\portable_data\cache\Other\BBC News - BBC News HD - 2019-03-31 16-00.ts","wb")
playing = False
while True:
  data = p.stdout.read(1000000)
  if data:
      video.write(data)
  else:
      break
video.close()
xbmcgui.Dialog().notification("Recording finished: BBC News HD", "BBC News", sound=True)
post_cmd = ['c:\\windows\\system32\\notepad.exe', 'D:\\kodi-18.1-Leia-x64\\portable_data\\cache\\Other\\BBC News - BBC News HD - 2019-03-31 16-00.ts']
p = subprocess.Popen(post_cmd, shell=True)
```

**Comment by deadlocker8** [comment]
Yes, I have. I already recognized your break statement and it's also generated for my jobs.
If the ffmpeg process is stopping then your code works totally fine (as I told about the LibeELEC system).
But on Windows the ffmpeg process is never stopped and continue streaming even if the rule's end time is already over for more than 10 minutes. I've already checked the pid file and it seems to contain the correct pid of the parent cmd.exe (holding the ffmpeg.exe process) process. I don't have any idea why this process is not killed after the end time is reached.

**Comment by primaeval** [comment]
It might be a problem with the UWP version of Kodi from the Microsoft Store. There are stricter permissions with accessing other processes and directories than with Windows native programs.
The -t parameter for ffmpeg controls how long it records in seconds. So it should time out, close and trigger the no data break.

**Comment by deadlocker8** [comment]
I also thought about this restriction. So checking the last job script for the `-t` parameter of ffmpeg reveals my fault. 
For testing this issue I created a manually timed trigger directly from the channel list in IPTV simple client setting the start and end time. For the time span I choose one minute. So for the ffmpeg `-t` option I expected `-t 60` or something like this. Checking the value in the generated py file reveals `-t 634` this means about 10 minutes which reminds me of the `Minutes After` option in the Jobs&Rules section of the plugin's settings where I set 10 minutes. If it's inteded that the amount of minutes specified in the  `Minutes After option` is added to manually rules as well, then everything is fine. Thanks again for your fast fix concerning the loop break statement.
